samza-core/src/main/java/org/apache/samza/task/TaskFactoryUtil.java

```diff
@@ -169,7 +169,10 @@ public static StreamApplication createStreamApplication(Config config) {
         Class<?> builderClass = Class.forName(appClassName);
         return (StreamApplication) builderClass.newInstance();
       } catch (Throwable t) {
-        throw new ConfigException(String.format("%s is not a StreamApplication.", appClassName));
+        String errorMsg = String.format("Failed to create StreamApplication class from the config. %s = %s",
+            StreamApplication.APP_CLASS_CONFIG, config.get(StreamApplication.APP_CLASS_CONFIG));
+        log.error(errorMsg, t);
```

Comment:
Slightly err on the side of double logs instead of missed logs. Run time exceptions like these are not handled in any caller. Hence, would be nice to log this here anyways.